### ISSUE TITLE
Memory consumption issue and solution

### ISSUE THREAD

**Issue by echeloni11** [TeaPearce/Conditional_Diffusion_MNIST#12]
Hi, 

Thank you so much for the code!

I am working with this code these days and I found that sampling from a trained model consumes significant memory. I looked into this problem and I found that it is helpful to add a line "x_i = x_i.detach()" at the end of the for loop in function DDPM.sample in script.py.

I hope it helps!

Thanks,
Harry
